app/assets/javascripts/components/high_order/expandable.jsx

```diff
@@ -22,6 +24,13 @@ const Expandable = function (Component) {
       return UIActions.open(this.refs.component.getKey());
     },
 
+    handleClickOutside() {
+      // TODO: Fire UIActions.close?
```

Comment:
It's poorly named, but `UIActions.open` actually toggles the state of it. (See how UIStore handles the `OPEN_KEY` signal for the details.)

With this version, it closes the popover, but the UIStore still registers it as open, so it will take two clicks to open it again. This function also tries to set state on every outside click, even when the popover is not open.

If you change the setState to instead call `this.open(e)`, and only if it's already open, then this seems to work perfectly.